torch/onnx/symbolic.py

```diff
@@ -173,11 +177,30 @@ def sigmoid(g, self):
 
 
 def mean(g, self, dim=None, keepdim=None):
-    kwargs = {}
+    if dim is None and keepdim is None:
+        return g.op("Mean", self)
     # NB: ONNX's default is different from PyTorch's
     if keepdim is None:
         keepdim = 0
-    return g.op("ReduceMean", self, axes_i=dim, keepdims_i=keepdim)
+    return g.op("ReduceMean", self, axes_i=[dim], keepdims_i=keepdim)
+
+
+def sum(g, self, dim=None, keepdim=None):
+    if dim is None and keepdim is None:
+        return g.op("Sum", self)
+    if keepdim is None:
+        keepdim = 0
+    return g.op("ReduceSum", self, axes_i=[dim], keepdims_i=keepdim)
+
+
+def prod(g, self, dim=None, keepdim=None):
```

Comment:
The ATen prod is not enabled yet.

https://github.com/ezyang/pytorch-1/blob/bc75b1cd59422802929ee310f7bed6c07d090904/tools/autograd/derivatives.yaml#L390